sormas-backend/src/main/java/de/symeda/sormas/backend/report/AggregateReportFacadeEjb.java

```diff
@@ -307,6 +310,85 @@ public List<AggregatedCaseCountDto> getIndexList(AggregateReportCriteria criteri
 		return resultList;
 	}
 
+	@Override
+	public List<AggregateReportDto> getDuplicateAggregateReports(AggregateReportCriteria criteria) {
+
+		CriteriaBuilder cb = em.getCriteriaBuilder();
+		CriteriaQuery<AggregateReport> cq = cb.createQuery(AggregateReport.class);
+		Root<AggregateReport> root = cq.from(AggregateReport.class);
+
+		Predicate filter = service.createUserFilter(cb, cq, root);
+		if (criteria != null) {
+			Predicate criteriaFilter = service.createCriteriaFilter(criteria, cb, cq, root);
+			filter = CriteriaBuilderHelper.and(cb, filter, criteriaFilter);
+		}
+
+		if (filter != null) {
+			cq.where(filter);
+		}
+
+		List<AggregateReport> resultList = em.createQuery(cq).getResultList();
+
+		List<AggregateReportDto> aggregateReportDtoList =
+			resultList.stream().map(aggregateReport -> toDto(aggregateReport)).collect(Collectors.toList());
+
+		aggregateReportDtoList.sort(
+			Comparator.comparing(AggregateReportDto::getDisease, Comparator.nullsFirst(Comparator.comparing(Disease::toString)))
+				.thenComparing(AggregateReportDto::getYear, Comparator.nullsFirst(Comparator.naturalOrder()))
+				.thenComparing(AggregateReportDto::getEpiWeek, Comparator.nullsFirst(Comparator.naturalOrder()))
+				.thenComparing(
+					r -> r.getAgeGroup() != null
+						? r.getAgeGroup().split("_")[0].replaceAll("[^a-zA-Z]", StringUtils.EMPTY).toUpperCase()
+						: StringUtils.EMPTY)
+				.thenComparing(
+					r -> r.getAgeGroup() != null ? Integer.parseInt(r.getAgeGroup().split("_")[0].replaceAll("[^0-9]", StringUtils.EMPTY)) : 0)
+				.thenComparing(AggregateReportDto::getRegion, Comparator.nullsFirst(Comparator.naturalOrder()))
+				.thenComparing(AggregateReportDto::getDistrict, Comparator.nullsFirst(Comparator.naturalOrder()))
+				.thenComparing(AggregateReportDto::getHealthFacility, Comparator.nullsFirst(Comparator.naturalOrder()))
+				.thenComparing(AggregateReportDto::getPointOfEntry, Comparator.nullsFirst(Comparator.naturalOrder())));
+
+		List<AggregateReportDto> onlyDuplicatesReturnList = new ArrayList<>();
+
+		aggregateReportDtoList.forEach(aggregateReportDto -> {
```

Comment:
I believe this whole part of the code could be simplified by something like this:

```
aggregateReportDtoList.forEach(r -> {
  List<AggregateReportDto> duplicateReports = aggregateReportDtoList.stream().filter(r2 -> !r.equals(r2) && isDuplicate(r, r2));
  if (duplicateReports.stream.anyMatch(r2 -> r.getChangeDate().before(r2.getChangeDate()))) {
    r.setDuplicate(true);
  }
  onlyDuplicatesReturnList.add(r);
  onlyDuplicatesReturnList.addAll(duplicateReports);
}
```

Would of course require to turn onlyDuplicatesReturnList to a Set and convert that set back to a list when returning it, or alternatively doing contains checks. Correct me if I'm wrong, maybe I'm missing something here, or whether you think that my approach doesn't make much sense (absolutely possible of course :D).

Reply:
with some additional changes I updated the logic